src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -678,4 +678,28 @@ object messages {
            |$fixedVarInAlternative""".stripMargin
     }
   }
+
+  case class IdentifierExpected(identifier: String)(implicit ctx: Context) extends Message(24) {
+    val kind = "Syntax"
+
+    val msg = "identifier expected"
+
+    val wrongIdentifier = s"def foo: $identifier = {...}"
+
+    val validIdentifier = s"def foo = {...}"
+
+    val explanation = {
+      hl"""|A valid identifier is expected, but $identifier was found.
```

Comment:
maybe quotes around `$identifier` so the line becomes:

``hl"""|A valid identifier expected, but `$identifier` found.``